diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/TopLevelOrderRule.kt

```diff
@@ -0,0 +1,124 @@
+import org.cqfn.diktat.ruleset.utils.*
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.CLASS
+import com.pinterest.ktlint.core.ast.ElementType.FILE
+import com.pinterest.ktlint.core.ast.ElementType.FUN
+import com.pinterest.ktlint.core.ast.ElementType.IMPORT_LIST
+import com.pinterest.ktlint.core.ast.ElementType.INTERNAL_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.OBJECT_DECLARATION
+import com.pinterest.ktlint.core.ast.ElementType.OVERRIDE_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.PRIVATE_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.PROPERTY
+import com.pinterest.ktlint.core.ast.ElementType.PROTECTED_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.WHITE_SPACE
+import com.pinterest.ktlint.core.ast.isPartOfComment
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.KtFunction
+import org.jetbrains.kotlin.psi.psiUtil.isExtensionDeclaration
+import org.jetbrains.kotlin.psi.psiUtil.siblings
+
+/**
+ * Rule that checks order in top level
+ */
+class TopLevelOrderRule(private val configRules: List<RulesConfig>) : Rule("top-level-order") {
+    private var isFixMode: Boolean = false
+    private lateinit var emitWarn: EmitType
+
+    override fun visit(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: EmitType
+    ) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (node.elementType == FILE) {
+            checkNode(node)
+        }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun checkNode(node: ASTNode) {
+        val children = node.getChildren(null)
+        val initialElementsOrder = children.filter { it.elementType in sortedType }
+        if (initialElementsOrder.isEmpty()) {
+            return
+        }
+        val properties = Properties(children.filter { it.elementType == PROPERTY }).sortElements()
+        val functions = children.filter { it.elementType == FUN }
+        val classes = children.filter { it.elementType == CLASS || it.elementType == OBJECT_DECLARATION }
+        val sortOrder = Blocks(properties, functions, classes).sortElements().map { astNode ->
+            Pair(astNode, astNode.siblings(false).takeWhile { it.elementType == WHITE_SPACE || it.isPartOfComment() }.toList())
+        }
+        val lastNonSortedChildren = initialElementsOrder.last().siblings(true).toList()
+        sortOrder.filterIndexed { index, pair -> initialElementsOrder[index] != pair.first }
+            .forEach { listOfChildren ->
+                val wrongNode = listOfChildren.first
+                TOP_LEVEL_ORDER.warnAndFix(configRules, emitWarn, isFixMode, wrongNode.text, wrongNode.startOffset, wrongNode) {
+                    node.removeRange(node.findChildByType(IMPORT_LIST)!!.treeNext, node.lastChildNode)
+                    node.removeChild(node.lastChildNode)
+                    sortOrder.map { bodyChild ->
```

Comment:
```suggestion
                    sortOrder.map { (..., ...) ->
```
Maybe, use destructuring and more readable names?